template.yaml

```diff
@@ -1242,3 +1242,99 @@ Resources:
       LogGroupName: !Sub "/aws/lambda/${Environment}-${AWS::StackName}-method-management-update-v1"
       RetentionInDays: 30
       KmsKeyId: !GetAtt LoggingKmsKey.Arn
+
+  MethodManagementDeletev1ApiStubFunction:
+    Type: AWS::Serverless::Function
+    # checkov:skip=CKV_AWS_116
+    DependsOn:
+      - MethodManagementDeletev1ApiStubFunctionLogGroup
+    Properties:
+      FunctionName: !Sub ${Environment}-${AWS::StackName}-method-management-delete-v1
+      Architectures: ["arm64"]
+      CodeUri: src
+      Handler: method-management.deleteMethodHandler
+      KmsKeyArn: !GetAtt LambdaKMSKey.Arn
+      PackageType: Zip
+      MemorySize: 128
+      ReservedConcurrentExecutions: 30
+      Runtime: nodejs18.x
+      Role: !GetAtt MethodManagementDeletev1ApiStubFunctionRole.Arn
+      Timeout: 5
+      Events:
+        userinfo:
+          Type: Api
+          Properties:
+            Path: /v1/mfa-methods/{mfaIdentifier}
+            Method: DELETE
+            RestApiId:
+              Ref: MethodManagementv1StubsRestApi
+      VpcConfig:
+        SubnetIds:
+          - Fn::ImportValue: !Sub ${VpcStackName}-PrivateSubnetIdA
+          - Fn::ImportValue: !Sub ${VpcStackName}-PrivateSubnetIdB
+        SecurityGroupIds:
+          - Fn::ImportValue: !Sub ${VpcStackName}-AWSServicesEndpointSecurityGroupId
+      Environment:
+        Variables:
+          TABLE_NAME: !Ref OidcStubDataTableName
+    Metadata:
+      BuildMethod: esbuild
+      BuildProperties:
+        Minify: true
+        Target: "es2020"
+        Sourcemap: true
+        EntryPoints:
+          - method-management/method-management.ts
+
+  MethodManagementDeletev1ApiStubFunctionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      PermissionsBoundary: !If
+        - UsePermissionsBoundary
+        - !Ref PermissionsBoundary
+        - !Ref AWS::NoValue
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole
+        - !Ref MethodManagementDeletev1ApiStubPolicy
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - "sts:AssumeRole"
+
+  MethodManagementDeletev1ApiStubPolicy:
+    Type: AWS::IAM::ManagedPolicy
```

Comment:
Not for this PR, but all the method management lambdas need the same permissions. I bet we could have a single role all the lambdas run as with a single policy for the DynamoDB permissions and save ourselves a lot of code in this template. 

I'll have a go at that later.

Reply:
Was much simpler than I thought: https://github.com/govuk-one-login/account-management-stubs/pull/205